hs_core/views/__init__.py

```diff
@@ -1719,6 +1707,41 @@ def _set_resource_sharing_status(user, resource, flag_to_set, flag_value):
     return None
 
 
+class FindGroupsView(TemplateView):
+    template_name = 'pages/groups-unauthenticated.html'  # default view is for users not logged in
+
+    def dispatch(self, *args, **kwargs):
+        if self.request.user.is_authenticated():
+            self.template_name = 'pages/groups-authenticated.html'  # update template if user is logged in
```

Comment:
oh, I see what you're doing now.... I don't understand why you can't just use the same template though.

Reply:
I think you could save yourself from maintaining duplicate code in the authenticated/unauthenticated views.  Seems like they should be mostly the same and the authenticated should have just a bit extra than the unauthenticated.

Reply:
The same template cannot be used because of how the feature will be rolled out, with a LOCAL_SETTINGS cutover.